query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -35,14 +39,62 @@
   // while the selectedColumns
   // is a set of column names.
   private final LinkedHashSet<Expression> allSelections;
+  private final Optional<Double> timeSeriesPeriod;
+  private final double timeRangeDuration;
 
   public ExecutionContext(String tenantId, QueryRequest request) {
     this.tenantId = tenantId;
     this.selectedColumns = new LinkedHashSet<>();
     this.allSelections = new LinkedHashSet<>();
+    this.timeSeriesPeriod = setTimeSeriesPeriod(request);
+    this.timeRangeDuration = setTimeRangeDuration(request);
     analyze(request);
   }
 
+  private Optional<Double> setTimeSeriesPeriod(QueryRequest request) {
+    Optional<Double> period = Optional.empty();
+    if (request.getGroupByCount() > 0) {
+      for (Expression expression : request.getGroupByList()) {
+        if (expression.getValueCase() == ValueCase.FUNCTION
+            && expression.getFunction().getFunctionName().equals("dateTimeConvert")) {
+          String periodInIso =
+              expression
+                  .getFunction()
+                  .getArgumentsList()
+                  .get(3)
+                  .getLiteral()
+                  .getValue()
+                  .getString();
+          period = Optional.of((double) isoDurationToSeconds(periodInIso));
+        }
+      }
+    }
+    return period;
+  }
+
+  private double setTimeRangeDuration(QueryRequest request) {
+    Optional<Long> duration = Optional.empty();
```

Comment:
This is always specified in milli seconds I suppose. Do we need duration for this ?

Reply:
Though we can store the output in Duration format.

Reply:
> Do we need duration for this ?
if we're representing a duration of time in code, primitives give us no benefit - they strictly lead to potential ambiguity on units that needs to be conveyed through naming, and are overall harder to use. So rather than asking if we need a duration - we should aim for that unless compelling reasons otherwise.

> This is always specified in milli seconds I suppose
Not necessarily, and that's why we want to get away from _any_ assumptions about units. One of our pinot views (I think it's log events?) made the switch to nanos, and we should support others doing the same.